rustsec/src/repository/git/commit.rs

```diff
@@ -40,40 +37,64 @@ pub struct Commit {
 impl Commit {
     /// Get information about HEAD
     pub(crate) fn from_repo_head(repo: &Repository) -> Result<Self, Error> {
-        let head = repo.repo.head()?;
+        let commit = repo
+            .repo
+            .head_commit()
+            .map_err(|err| format_err!(ErrorKind::Repo, "unable to locate head commit: {}", err))?;
 
-        let oid = head.target().ok_or_else(|| {
+        // Since we are pulling multiple pieces from the commit it's better to do this once
+        let cref = commit.decode().map_err(|err| {
             format_err!(
                 ErrorKind::Repo,
-                "no ref target for: {}",
-                repo.path.display()
+                "unable to decode commit information: {}",
+                err
             )
         })?;
 
-        let commit_id = oid.to_string();
-        let commit_object = repo.repo.find_object(oid, Some(git2::ObjectType::Commit))?;
-        let commit = commit_object.as_commit().unwrap();
-        let author = commit.author().to_string();
-
-        let summary = commit
-            .summary()
-            .ok_or_else(|| format_err!(ErrorKind::Repo, "no commit summary for {}", commit_id))?
-            .to_owned();
-
-        let (signature, signed_data) = match repo.repo.extract_signature(&oid, None) {
-            Ok((ref sig, ref data)) => {
-                (Some(Signature::from_bytes(sig)?), Some(data.deref().into()))
-            }
-            _ => (None, None),
+        let commit_id = commit.id;
+        let timestamp = crate::repository::git::gix_time_to_time(cref.committer.time);
+        let author = {
+            let sig = cref.author();
+            format!("{} <{}>", sig.name, sig.email)
         };
 
-        let time = UNIX_EPOCH + Duration::from_secs(commit.time().seconds() as u64);
+        let summary = cref.message_summary().to_string();
+
+        if summary.is_empty() {
+            return Err(format_err!(
+                ErrorKind::Repo,
+                "no commit summary for {}",
+                commit_id
+            ));
+        }
+
+        let (signature, signed_data) = if let Some(sig) = cref.extra_headers().pgp_signature() {
```

Comment:
As noted, I opened https://github.com/Byron/gitoxide/pull/971 to add this to gix itself, but in the meantime this is just a more inefficient way to get the signature/signed data. This could also just be removed since it's not currently validated at all.

Reply:
This is now available via [this PR](https://github.com/Byron/gitoxide/pull/973) will be called `commit.signature()`. Basic validation seems straightforward if `gpg` can be used as well.